### ISSUE TITLE
Compile list of syscalls required by wasmtime functions

### ISSUE THREAD

**Issue by MikeCamel** [enarx/enarx#694]
**Description**
In order to know what syscalls we need to implement in the various shims, we need a list of the syscalls called by wasmtime (our chosen WASI runtime).  There is a test suite here: [https://github.com/bytecodealliance/wasmtime/tree/main/crates/test-programs/wasi-tests](https://github.com/bytecodealliance/wasmtime/tree/main/crates/test-programs/wasi-tests).  This is a good first issue, as it requires the set-up of the appropriate dev environment (rust with the target `wasm32-wasi`, wasmtime binary, etc.), and encourages an understanding of why we need this within the architecture.

Sub-tasks:

1. set-up dev environment
1. compile the tests
1. check each test with wasmtime to ensure they run (some have usage requirements)
1. run each test with strace -c, e.g. `strace -c testtesttest.wasm &>> syscalls.txt`
1. take the resulting concatenated file and collate to a list of all required syscalls
1. if you're feeling really keen, sort by both syscall number and syscall name
1. record these somewhere that the developer team can prioritise them

**Comment by ziyi-yan** [comment]
Hi, @MikeCamel, I'm working on setting up dev environment recently. And I just compiled and tried out some wasm programs from `wasi-tests`.

Feel free to assign me to this issue. I would create a meta issue with a todo-list of those syscalls I collated for further prioritizing.

**Comment by MikeCamel** [comment]
Assigned to Ziyi.  If it's easy, please create separate meta-issues for the different tests, or group the syscalls in logical sets.  Thanks!

**Comment by ziyi-yan** [comment]
@MikeCamel Here is [the meta issue](https://github.com/enarx/enarx/issues/733) I put all the syscalls required by `wasi-tests` running on wasmtime in Linux.

I also link the script in the description, which I used for generating the syscall list. I found out that most of the `wasi-tests` program required a scratch pre-opened directory, so I passed all the programs with a scratch directory. @kubkon @ueno Let me know if I misuse any test programs or miss any syscalls.

If this step is done correctly, I will continue to group them into multiple logical sets.
